server/src/WebSocketEvents.js

```diff
@@ -290,20 +290,89 @@ eventResponses.mv_r = (clientSocket) => {
  * @param {*} clientSocket
  * @param {String} data
  */
+
+// TODO: Not sure where to put this
+let entityChatCooldownList = [
+    // { id: 1, scope: "GLOBAL" },
+];
+
 eventResponses.chat = (clientSocket, data) => {
-    if (!data) return;
+    // Can't use ChatState here 'cause it's located outside module
+    const CHAT_SCOPES = {
+        LOCAL: {
+            value: "LOCAL",
+            cooldown: 0,
+        },
+        GLOBAL: {
+            value: "GLOBAL",
+            cooldown: 10000,
+        },
+        TRADE: {
+            value: "TRADE",
+            cooldown: 10000,
+        },
+    };
```

Comment:
I would prefer these to be in the server settings file `settings.example.yml` for easier configuration.

```yml
# The channels/categories of chat topics that players can send their messages in, with configurable
# cooldowns to limit the message rate of each player in that scope.
CHAT_SCOPES:
  - name: "LOCAL"
    cooldown: 0
  - name: "GLOBAL"
    cooldown: 0
  - name: "TRADE"
    cooldown: 0

# How long incoming player chat messages can be.
MAX_CHAT_MESSAGE_LENGTH: 255
```

Then they can be accessed as `settings.CHAT_SCOPES`
i.e.
```js
console.log("chat scopes:", settings.CHAT_SCOPES);
/*
chat scopes: [
    { name: 'LOCAL', cooldown: 0 },
    { name: 'GLOBAL', cooldown: 0 },
    { name: 'TRADE', cooldown: 0 }
]
*/
```
then you can use/reformat the values however you need.